tests/scripts/all.sh

```diff
@@ -1527,10 +1522,82 @@ component_test_full_no_cipher () {
     scripts/config.py unset MBEDTLS_LMS_PRIVATE
     make
 
-    msg "test: full minus CIPHER"
+    msg "test: full no CIPHER no PSA_CRYPTO_C"
     make test
 }
 
+# This is a common configurator and test function that is used in:
+# - component_test_full_no_cipher_with_crypto
+# - component_test_full_no_cipher_with_crypto_config
+# It accepts 2 input parameters:
+# - $1: boolean value which basically reflects status of MBEDTLS_PSA_CRYPTO_CONFIG
+# - $2: a text string which describes the test component
+common_test_full_no_cipher_with_crypto () {
+    USE_CRYPTO_CONFIG="$1"
+    COMPONENT_DESCRIPTION="$2"
+
+    msg "build: $COMPONENT_DESCRIPTION"
+
+    scripts/config.py full
+    scripts/config.py unset MBEDTLS_CIPHER_C
+
+    if [ "$USE_CRYPTO_CONFIG" -eq 1 ]; then
+        # The built-in implementation of these modes currently depends on CIPHER_C
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CCM
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CCM_STAR_NO_TAG
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_GCM
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CMAC
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CBC_NO_PADDING
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CBC_PKCS7
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CFB
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CTR
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_ECB_NO_PADDING
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_OFB
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_STREAM_CIPHER
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_AES
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_DES
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_CAMELLIA
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_ARIA
+    else
+        # Don't pull in cipher via PSA mechanisms
+        scripts/config.py unset MBEDTLS_PSA_CRYPTO_CONFIG
+    fi
+    # The following modules directly depends on CIPHER_C
+    scripts/config.py unset MBEDTLS_CCM_C
+    scripts/config.py unset MBEDTLS_CMAC_C
+    scripts/config.py unset MBEDTLS_GCM_C
+    scripts/config.py unset MBEDTLS_NIST_KW_C
+    scripts/config.py unset MBEDTLS_PKCS12_C
+    scripts/config.py unset MBEDTLS_PKCS5_C
+    scripts/config.py unset MBEDTLS_SSL_TLS_C
+    scripts/config.py unset MBEDTLS_SSL_TICKET_C
+    # Disable cipher modes/keys that make PSA depend on CIPHER_C.
```

Comment:
Actually I don't think that's the right place for those: IMO they should be in the "else" part of the `if [ "$USE_CRYPTO_CONFIG" -eq 1 ]` above: when using PSA-based config, we need to disable PSA symbols, and when using legacy-based config, we need to disable legacy symbols (otherwise, the corresponding PSA features would get enabled, making PSA depend on cipher).

Or am I missing some way the legacy options still influence PSA options even when `PSA_CRYPTO_CONFIG` is enabled?

Reply:
I'm not sure: `mbedtls_config.h` is evaluated independently of the status of `PSA_CRYPTO_CONFIG`. The `adjust` header files (either legacy->PSA or PSA->legacy) are additive, not subtractive. So if for example we have `MBEDTLS_CCM_C` enabled in `mbedtls_config.h` then it will still be enabled even if `PSA_CRYPTO_CONFIG` is evaluated and `PSA_WANT_ALG_CCM` is disabled. 
Am I wrong?

Reply:
For CCM and GCM, yes, sure, but they're already disabled in the section "depens on `CIPHER_C`" as they should.

Here I was talking about "cipher modes/keys that make PSA depend on CIPHER_C" as the comment says. My point is:
- we don't need to disable these in the `full_no_cipher` component that doesn't have PSA;
- so we only disable them because of the influence they could have on PSA;
- unless I'm missing something, they should only influence PSA when `PSA_CRYPTO_CONFIG` is disabled;
- so, we should only need to disable them when `PSA_CRYTO_CONFIG` is disabled.

Reply:
Sorry in my first comment I erroneously associated your comment to the wrong line.
Then, looking at the 2nd comment, I was like: "Ok, even if it builds, then wouldn't it be better to still disable all the `MBEDTLS` symbols for modes/key-pairs as well. They are not used so what's the purpose for keeping them there?". But then I tried and found that there were 3 components in `test_suite_psa_crypto` which [were failing](https://github.com/Mbed-TLS/mbedtls/pull/8119/commits/1ce6f0fabbfa15bce6c5a42f5cc799ee4ca7a422) due to wrong dependencies in the test. So this change was really beneficial in the end :)

Reply:
I think as a general rule of thumb, it's always better to keep as many things enabled as possible, in order to get better test coverage. The obvious exception being tests that are focused on how much you can disable while keeping (some) things working :)